GoogleDataTransport/GDTCORLibrary/GDTCORLifecycle.m

```diff
@@ -47,18 +47,17 @@ - (instancetype)init {
   if (self) {
     NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
     [notificationCenter addObserver:self
-                           selector:@selector(applicationDidEnterBackground:)
+                           selector:@selector(applicationDidEnterBackgroundNotification:)
                                name:kGDTCORApplicationDidEnterBackgroundNotification
                              object:nil];
     [notificationCenter addObserver:self
-                           selector:@selector(applicationWillEnterForeground:)
+                           selector:@selector(applicationWillEnterForegroundNotification:)
                                name:kGDTCORApplicationWillEnterForegroundNotification
                              object:nil];
 
-    NSString *name = kGDTCORApplicationWillTerminateNotification;
-    [notificationCenter addObserver:self
-                           selector:@selector(applicationWillTerminate:)
-                               name:name
+    [notificationCenter addObserver:kGDTCORApplicationWillTerminateNotification
```

Comment:
I think, `kGDTCORApplicationWillTerminateNotification` should be `self` here.

Reply:
I'm glad we have it covered by tests since I overlooked it initially.

Reply:
🤦

Reply:
Given my track record on this CL, I'm going to leave the delegate removal alone for right now if that's ok...

Reply:
> Given my track record on this CL, I'm going to leave the delegate removal alone for right now if that's ok...

sure, no problem.